builtin/logical/pki/ca_util.go

```diff
@@ -314,6 +315,7 @@ func buildSignVerbatimRole(data *framework.FieldData, role *roleEntry) *roleEntr
 		}
 		entry.NoStore = role.NoStore
 		entry.Issuer = role.Issuer
+		entry.BasicConstraintsValidForNonCA = role.BasicConstraintsValidForNonCA
```

Comment:
Should this be added conditionally on whether the parameter was provided? 

If so, I'd do:

```go
if , ok := data.GetOk("basic_constraints_valid_for_non_ca"); !ok {
    entry.BasicConstraintsValidForNonCA = role.BasicConstraintsValidForNonCA
}
```

That way, if a role requires it, but the API call explicitly disables it, you respect that explicit disabling. My 2c.